jawt-docs/dev/core.md

```
@@ -0,0 +1,262 @@
+*   **Logging**: A structured logging interface (`Logger`) and a default implementation (`DefaultLogger`) provide consistent and informative output.
+*   **Paths**: The `ProjectPaths` struct centralizes the management and resolution of all file and directory paths within a JAWT project.
+
+## Key Data Structures
+
+### `JawtConfig`
+
+Represents the global JAWT configuration, typically loaded from `jawt.config.json` (though currently defaults are used if no file is present). It includes paths to external tools and general JAWT settings.
+
+```go
+type JawtConfig struct {
+	TypeScriptPath     string `json:"typescript_path"`
```

markdownlint-cli2 (0.17.2)

Hard tabs
Column: 1

(MD010, no-hard-tabs)